src/Swashbuckle.AspNetCore.Cli/HostingApplication.cs

```diff
@@ -31,6 +33,18 @@ void ConfigureHostBuilder(object hostBuilder)
                 {
                     services.AddSingleton<IServer, NoopServer>();
                     services.AddSingleton<IHostLifetime, NoopHostLifetime>();
+
+                    for (var i = services.Count - 1; i >= 0; i--)
+                    {
+                        // exclude all implementations of IHostedService
+                        // except Microsoft.AspNetCore.Hosting.GenericWebHostService because that one will build/configure
+                        // the WebApplication/Middleware pipeline in the case of the GenericWebHostBuilder.
+                        if (typeof(IHostedService).IsAssignableFrom(services[i].ServiceType)
```

Comment:
```suggestion
                        if (services[i].ServiceType == typeof(IHostedService)
```

This is taken from https://github.com/RicoSuter/NSwag/pull/4756

Reply:
Thanks! Do you have a specific example of a service that shouldn't be removed so I can augment the test?